src/citrine/resources/delete.py

```diff
@@ -71,31 +73,45 @@ def _async_gemd_batch_delete(
 
     if dataset_id is not None:
         body.update({'dataset_id': str(dataset_id)})
-
-    path = format_escaped_url('/projects/{project_id}/gemd/async-batch-delete',
-                              project_id=project_id
-                              )
+    if team_id is not None:
+        path = format_escaped_url('/teams/{team_id}/gemd/async-batch-delete',
+                                  team_id=team_id)
+    elif project_id is not None:
+        warn("project_id is deprecated as of 3.4.0, and will be removed in 4.0.0. Please provide "
+             "team_id instead",
+             DeprecationWarning)
```

Comment:
Why are we issuing a Deprecation warning inside a private method?

Reply:
Looking more closely, I don't think Lenore knew about `@deprecated`, and was actually trying to deprecate the place we call it from. I'll make that change.